Mathlib/Lean/Meta/Basic.lean

```diff
@@ -52,3 +53,18 @@ def Lean.Meta.forallMetaTelescopeReducingUntilDefEq
     bis := bis ++ bs
     out := tp
   return (mvs, bis, out)
+
+/-- Determines whether two expressions are definitionally equal to each other without any mvar
+    assignments. -/
```

Comment:
I'm not so sure about this, since it's just doing what the `isDefEq` documentation suggests doing, and it's just saving a few characters while adding another function to the API.

If you do want it, we need to make the documentation very clear about what it does.

```suggestion
/-- `pureIsDefEq e₁ e₂` is short for `withNewMCtxDepth <| isDefEq e₁ e₂`.
Determines whether two expressions are definitionally equal to each other
when metavariables are not assignable. -/
```